corehq/util/view_utils.py

```diff
@@ -129,3 +133,23 @@ def reverse(viewname, params=None, absolute=False, **kwargs):
 
 def absolute_reverse(*args, **kwargs):
     return reverse(*args, absolute=True, **kwargs)
+
+
+def get_case_or_404(domain, case_id):
+    try:
+        case = CaseAccessors(domain).get_case(case_id)
+        if case.domain != domain or case.is_deleted:
+            raise Http404()
+        return case
+    except CaseNotFound:
+        raise Http404()
+
+
+def get_form_or_404(domain, id):
+    try:
+        form = FormAccessors(domain).get_form(id)
+        if form.domain != domain:
```

Comment:
should this have a similar `form.is_deleted` check?

Reply:
Seems reasonable: e532ad7